fmmax/sources.py

```diff
@@ -38,22 +38,28 @@ def amplitudes_for_fields(
     Returns:
         The forward and backward wave amplitudes.
     """
-    if not (ex.shape == ey.shape == hx.shape == hy.shape) or ex.ndim != 3:
+
+    batch_shape = layer_solve_result.batch_shape
+    expected_fields_dims = (
```

Comment:
Maybe give this comment its own line, so that this formatting is a bit less awkward?